src/ext/oblivc/obliv_bits.c

```diff
@@ -307,22 +306,23 @@ static ProtocolTransport* tcp2PSplit(ProtocolTransport* tsrc)
 {
   tcp2PTransport* t = CAST(tsrc);
   fflush(t->sockStream); 
-#ifdef PROFILE_NETWORK
-  ((tcp2PTransport*)t)->flushCount++;
-#endif
   // I should really rewrite sockSplit to use FILE* sockStream
   int newsock = sockSplit(t->sock,tsrc,t->isClient);
   if(newsock<0) { fprintf(stderr,"sockSplit() failed\n"); return NULL; }
-#ifdef PROFILE_NETWORK
-  if(!t->isClient) t->bytes+=sizeof(in_port_t);
-#endif
   tcp2PTransport* tnew = tcp2PNew(newsock,t->isClient);
-#ifdef PROFILE_NETWORK
   tnew->parent=t;
-#endif
   return CAST(tnew);
 }
 
+static ProtocolTransport* tcp2PSplitProfiled(ProtocolTransport* tsrc)
+{
+  tcp2PTransport* t = CAST(tsrc);
+  ((tcp2PTransport*)t)->flushCount+=2;
+  if(!t->isClient) t->bytes+=sizeof(in_port_t);
+  
+  return tcp2PSplit(tsrc);
```

Comment:
Agreed. We could use the oflush function that I added, which would take care of the profiled case automatically, but we'd have to be sure that no one does a plain fwrite (as opposed to osend).